src/fheroes2/battle/battle_interface.cpp

```diff
@@ -2927,7 +2933,32 @@ void Battle::Interface::HumanBattleTurn( const Unit & unit, Actions & actions, s
         }
     }
     else if ( le.MouseCursor( battleFieldRect ) ) {
-        const int themes = GetBattleCursor( msg );
+        int themes = GetBattleCursor( msg );
+
+        if ( _swipeAttack.isValid ) {
+            // The swipe attack motion is either in progress or has finished.
+            if ( index_pos == _swipeAttack.swipeDstCellIndex ) {
+                // The cursor is above the stored destination, we should display the stored attack theme.
+                themes = _swipeAttack.swipeDstTheme;
+            }
+            else {
+                // The cursor has left the destination. Abort the swipe attack.
+                _swipeAttack = {};
+                _boardActionIntent = {};
+            }
+        }
+        else if ( _swipeAttack.isValidSwipeAttackDestination( themes, index_pos, _currentUnit->GetHeadIndex(), _currentUnit->GetTailIndex() ) ) {
+            // Valid swipe attack target cell. Calculate the attack angle based on destination and source cells.
+            int direction = Board::GetDirection( index_pos, _swipeAttack.swipeSrcCellIndex );
+            themes = GetSwordCursorDirection( direction );
```

Comment:
Since the `direction` it is not used anywhere except for this single line, we can avoid this temporary variable:

```suggestion
            themes = GetSwordCursorDirection( Board::GetDirection( index_pos, _swipeAttack.swipeSrcCellIndex ) );
```